### PR TITLE
Fixes #14293: ncf 4.1 is not compatible anymore with agent 3.1

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/14293

### CODE REVIEW

File: tree/20_cfe_basics/dry_run_lib.unsupported.cf [hunk 1]
```diff
@@ -44,3 +44,18 @@ bundle agent set_dry_run_mode(mode)
     debug.unset_mode::
       "${configuration.debug} Dry-run mode is not compatible with this agent, but only a disabling of dry-run has been requested. Ignoring.";
 }
+
+# Doesn't do anything - necessary for signature compatibility
+bundle agent dry_run_initialization {
+}
+
+
+bundle agent push_dry_run_mode(mode)
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.unsupported.cf]: I think we should call set_dry_run_mode to fail in this case

File: tree/20_cfe_basics/dry_run_lib.unsupported.cf [hunk 2]
```diff
@@ -44,3 +44,18 @@ bundle agent set_dry_run_mode(mode)
...
+{
+
+}
+
+bundle agent pop_dry_run_mode()
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.unsupported.cf]: here too

**Comment by ncharles** [comment]
Commit modified

File: tree/20_cfe_basics/dry_run_lib.unsupported.cf [hunk 3]
```diff
@@ -44,3 +44,30 @@ bundle agent set_dry_run_mode(mode)
...
+}
+
+
+bundle agent push_dry_run_mode(mode)
+{
+  methods:
+      "abort" usebundle => _abort("unsupported_dryrun", "Dry-run mode is not compatible with this agent. Stopping immediately to prevent any unwanted change."),
+                action  => immediate_ignore_dry_run;
+
+  reports:
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.unsupported.cf]: you can remove the reports here

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**